language/extensions/async/move-async-vm/tests/testsuite.rs

```diff
@@ -0,0 +1,403 @@
+use move_vm_types::gas_schedule::GasStatus;
+use std::{
+    cell::RefCell,
+    collections::{BTreeMap, BTreeSet, VecDeque},
+    path::{Path, PathBuf},
+    str::FromStr,
+};
+
+const TEST_ADDR: &str = "0x3";
+const SOURCE_DIRS: &[&str] = &[
+    "./tests/sources",
+    "../move-async-lib/sources",
+    "../../../move-stdlib/sources",
+];
+
+struct Harness {
+    module_cache: BTreeMap<Identifier, CompiledUnit>,
+    vm: AsyncVM,
+    actor_instances: Vec<(ModuleId, AccountAddress)>,
+    baseline: RefCell<String>,
+    resource_store: RefCell<BTreeMap<(AccountAddress, StructTag), Vec<u8>>>,
+}
+
+fn test_account() -> AccountAddress {
+    AccountAddress::from_hex_literal(TEST_ADDR).expect("valid test address")
+}
+
+fn test_runner(path: &Path) -> datatest_stable::Result<()> {
+    let target_module = path
+        .with_extension("")
+        .file_name()
+        .unwrap()
+        .to_string_lossy()
+        .to_string();
+    let deps = extract_test_directives(path, "// dep:")?;
+    let actors = extract_test_directives(path, "// actor: ")?;
+    let instances = extract_test_directives(path, "// instance: ")?;
+    let harness = Harness::new(
+        std::iter::once(target_module.clone())
+            .chain(deps.into_iter())
+            .collect(),
+        actors,
+        instances,
+    )?;
+    harness.run(&target_module)?;
+    let baseline_path = path.with_extension(EXP_EXT);
+    verify_or_update_baseline(baseline_path.as_path(), harness.baseline.borrow().as_str())?;
+    Ok(())
+}
+
+datatest_stable::harness!(test_runner, "tests/sources", r".*\.move");
+
+// ========================================================================================
+// Test execution
+
+impl Harness {
+    fn run(&self, _module: &str) -> anyhow::Result<()> {
+        let mut gas = GasStatus::new_unmetered();
+        let mut tick = 0;
+        // Publish modules.
+        let mut proxy = HarnessProxy { harness: self };
+        let mut session = self.vm.new_session(test_account(), 0, &mut proxy);
+        let mut done = BTreeSet::new();
+        for id in self.module_cache.keys() {
+            self.publish_module(&mut session, id, &mut gas, &mut done)?;
+        }
+        // Initialize actors
+        let mut mailbox: VecDeque<Message> = Default::default();
+        for (actor, addr) in self.actor_instances.clone() {
+            self.log(format!(
+                "actor 0x{} created from {}",
+                addr.short_str_lossless(),
+                actor.short_str_lossless()
+            ));
+            let result = {
+                let mut proxy = HarnessProxy { harness: self };
+                let session = self.vm.new_session(addr, 0, &mut proxy);
+                session.new_actor(&actor, addr, &mut gas)
+            };
+            self.handle_result(&mut mailbox, result);
+
+            // Put a start message for this actor into the mailbox.
+            let entry_point_id = Identifier::from_str("start")?;
```

Comment:
"start" is the name of init function?

Reply:
This is just a convention for this test suite. Moving forward, we have to see whether we need a default mechanism for `start`.